appmonitoring/ts/src/CertificateGenerator.ts

```diff
@@ -0,0 +1,170 @@
+        return (1001).toString(16) + Math.ceil(Math.random()*100); //Just creates a placeholder hex and randomly increments it with a number between 1 and 100
+    }
+
+    private static async GenerateSelfSignedCertificate(): Promise<WebhookCertData> {
+        const caCert: forge.pki.Certificate = forge.pki.createCertificate();
+        const keys = forge.pki.rsa.generateKeyPair(4096);
+        caCert.serialNumber = CertificateManager.randomHexSerialNumber();
+        caCert.publicKey = keys.publicKey;
+        caCert.privateKey = keys.privateKey;
+        const timeNowNum: number = Date.now()
+        caCert.validity.notBefore = new Date(timeNowNum - (5 * 60 * 1000)); //5 Mins ago
+        caCert.validity.notAfter = new Date(timeNowNum + (2 * 365 * 24 * 60 * 60 * 1000)); //2 Years from now
+
+        const attributes = [{
+            shortName: 'CN',
+            value: 'applicationinsights-ca'
+        }];
+        caCert.setSubject(attributes);
+        caCert.setIssuer(attributes);
+    
+        const extensions = [{
+            name: 'basicConstraints',
+            cA: true
+        },
+        {
+            name: 'subjectKeyIdentifier',
+            keyIdentifier: caCert.generateSubjectKeyIdentifier().getBytes(),
+        },
+        {
+            name: 'keyUsage',
+            keyCertSign: true,
+            cRLSign: true,
+            digitalSignature: true,
+            keyEncipherment: true,
+        }];
+
+        caCert.setExtensions(extensions);
+        caCert.sign(caCert.privateKey,forge.md.sha256.create());
+
+        const caCertResult: CACertData = {
+            certificate: forge.pki.certificateToPem(caCert),
+            serviceKey: forge.pki.privateKeyToPem(caCert.privateKey)
+        }
+
+        const host_attributes = [{
+            shortName: 'CN',
+            value: WebhookDNSEndpoint
+        }];
+    
+        const host_extensions = [{
+            name: 'basicConstraints',
+            cA: false
+        }, 
+        {
+            name: 'authorityKeyIdentifier',
+            keyIdentifier: caCert.generateSubjectKeyIdentifier().getBytes(),
+        }, 
+        {
+            name: 'keyUsage',
+            digitalSignature: true,
+            keyEncipherment: true
+        },
+        {
+            name: 'extKeyUsage',
+            serverAuth: true
+        }, 
+        {
+            name: 'subjectAltName',
+            altNames: [{ type: 2, value: WebhookDNSEndpoint }]
+        }];
+
+        const newHostCert = forge.pki.createCertificate();
+        const hostKeys = forge.pki.rsa.generateKeyPair(4096);
+
+        // Set the attributes for the new Host Certificate
+        newHostCert.publicKey = hostKeys.publicKey;
+        newHostCert.serialNumber = CertificateManager.randomHexSerialNumber();
+        newHostCert.validity.notBefore = new Date(timeNowNum - (5 * 60 * 1000)); //5 Mins ago
+        newHostCert.validity.notAfter = new Date(timeNowNum + (2 * 365 * 24 * 60 * 60 * 1000)); //2 Years from now
```

Comment:
Can expiry of this cert be later than the expiry of the underlying CA cert? Worth checking.